src/common/redis_module/ray_redis_module.cc

```diff
@@ -665,28 +606,109 @@ int TableAdd_RedisCommand(RedisModuleCtx *ctx,
   }
 }
 
-// This is a temporary redis command that will be removed once
-// the GCS uses https://github.com/pcmoritz/credis.
-int TableLookup_RedisCommand(RedisModuleCtx *ctx,
+int TableAppend_RedisCommand(RedisModuleCtx *ctx,
                              RedisModuleString **argv,
                              int argc) {
-  if (argc != 4) {
+  if (argc != 5) {
     return RedisModule_WrongArity(ctx);
   }
 
   RedisModuleString *prefix_str = argv[1];
+  RedisModuleString *pubsub_channel_str = argv[2];
   RedisModuleString *id = argv[3];
+  RedisModuleString *data = argv[4];
 
-  RedisModuleKey *key = OpenPrefixedKey(ctx, prefix_str, id, REDISMODULE_READ);
-  if (key == nullptr) {
-    return RedisModule_ReplyWithNull(ctx);
+  // Set the keys in the table.
+  RedisModuleKey *key = OpenPrefixedKey(ctx, prefix_str, id,
+                                        REDISMODULE_READ | REDISMODULE_WRITE);
+  size_t index = RedisModule_ValueLength(key);
+  RedisModule_ZsetAdd(key, index, data, NULL);
+  RedisModule_CloseKey(key);
+
+  // Publish a message on the requested pubsub channel if necessary.
+  TablePubsub pubsub_channel = ParseTablePubsub(pubsub_channel_str);
+  if (pubsub_channel != TablePubsub_NO_PUBLISH) {
+    // All other pubsub channels write the data back directly onto the channel.
+    return PublishTableAdd(ctx, pubsub_channel_str, id, data);
+  } else {
+    return RedisModule_ReplyWithSimpleString(ctx, "OK");
   }
-  size_t len = 0;
-  const char *buf = RedisModule_StringDMA(key, &len, REDISMODULE_READ);
+}
 
-  RedisModule_ReplyWithStringBuffer(ctx, buf, len);
+/// A helper function to create and finish a GcsTableEntry, based on the
+/// current value or values at the given key.
+void TableEntryToFlatbuf(flatbuffers::FlatBufferBuilder &fbb,
```

Comment:
It's common to order in-args before out-args; how about `TableEntryToFlatbuf (table_key, entry_id, fbb)`?

Reply:
What are in-args and out-args?